src/webhook/handlers/tutorial.js

```diff
@@ -0,0 +1,387 @@
+                weight: 'bold',
+                wrap: true,
+              },
+            ],
+          },
+          footer: {
+            type: 'box',
+            layout: 'horizontal',
+            contents: [
+              {
+                type: 'button',
+                action: createPostbackAction(
+                  buttonLabel,
+                  SIMULATE_FORWARDING_MESSAGE,
+                  displayText,
+                  data.sessionId,
+                  'TUTORIAL'
+                ),
+                style: 'primary',
+                color: '#ffb600',
+              },
+            ],
+          },
+        },
+      ],
+    },
+  };
+}
+
+/**
+ * @returns {object} Flex message object
+ */
+function createEndingMessage() {
+  const text = t`This is the end of the tutorial. Next time when you receive a suspicious message, don't hesitate to forward it to me! 🤗`;
+  const imageUrl =
+    'https://uc4e58e1c257cac3657ad4ca0a69.previews.dropboxusercontent.com/p/thumb/AA_IcLRYeJqCzpdE1ZynRVM_LOWI3vkKHByHswDrgZS9BmYfzPao--HbC5krsws4BroUuhF9jQyuXs12XWmO5SuBgPbygVSHWNQM1wrEM_Sa9M-Q6kGD_t_xaXRbNoF2MbBsJfQBSAwJ9XuJsAs2zt8XLURjLX2Kz54W0v6NUuEeHdfetQFlStTzvicxVc_ROqMAYbzXvaE92j1SGHtxJaPZYxDdr5hZhwttLAkXccCgMy052jxgct0DTnSqzO0-cVk6DXRY755iHN_loO_cVlegYQDKmySTfpFMfZp8S8aSoLKsnk2Gqur4f_PabPrWFoMA289DZZ7wRSzdbN8DRlO9cTHo9FoQY-SdcOrpWBq4u6was6uvbXaHeZbrY4ZWmvm8JbrJqQWwb-txBVL1RUzN/p.png?size=800x600&size_mode=3';
+
+  return {
+    type: 'flex',
+    altText: ellipsis(text, 300) + '\n' + FLEX_MESSAGE_ALT_TEXT,
+    contents: {
+      type: 'carousel',
+      contents: [createImageTextBubble(imageUrl, text)],
+    },
+  };
+}
+
+/**
+ * @param {object} data
+ * @returns {object[]} message objects
+ */
+function createMockReplyMessages(data) {
+  const GetReply = {
+    type: 'RUMOR',
+    reference:
+      'http://www.mygopen.com/2017/06/blog-post_26.html\n神奇的地瓜葉？搭配鮮奶遠離三高？謠言讓醫生說：有痛風或是腎臟不好的人要小心！',
+    text:
+      '基本上地瓜葉其實單吃就有效果，牛奶、豆漿可加可不加，民眾不用迷信。 三高或是糖尿病的患者還是要搭配醫生的治療，不能單靠吃地瓜葉就想將身體調養好，民眾千萬要注意。\n另外地瓜葉內還有鉀和鈉，對於有痛風或是腎臟不好的民眾反而會造成負擔，因此並不建議食用。',
+  };
+  const GetArticle = { replyCount: 1 };
+  data.selectedArticleId = '2sn80q5l5mzi0';
+
+  const replies = createReplyMessages(
+    GetReply,
+    GetArticle,
+    data.selectedArticleId
+  );
+  // put quickreply into last message
+  replies[replies.length - 1]['quickReply'] = {
+    items: [
+      createQuickReplyPostbackItem(
+        PROVIDE_PERMISSION_SETUP,
+        data.sessionId,
+        'TUTORIAL'
+      ),
+      createQuickReplyPostbackItem(
+        EXPLAN_CHATBOT_FLOW_AND_PROVIDE_PERMISSION_SETUP,
+        data.sessionId,
+        'TUTORIAL'
+      ),
+    ],
+  };
+  return replies;
+}
+
+/**
+ * @param {object} data
+ * @param {string} message
+ * @returns {object} Flex message object
+ */
+function createPermissionSetupDialog(data, message) {
+  const buttonLabel = t`Setup permission`;
+  const buttonUri = `${process.env.LIFF_URL}/liff/index.html?p=permission`;
```

Comment:
Hmm by permission settings I actually meant our existing `settings` page. That's why I also put welcome messages in settings LIFF.